### ISSUE TITLE
Links in wikis using short URLs are not recognized

### ISSUE THREAD

**Issue by Nicknag** [SolidCharity/exportMediaWiki2HTML#26]
The script only works if links between wiki pages look like `mywiki.org/index.php?title=Main_Page`. If short URLs (see https://www.mediawiki.org/wiki/Manual:Short_URL) are used in a wiki the links are not recognized by the script. This setup is very common for example in Wikipedia:

https://en.wikipedia.org/wiki/Main_Page

**Comment by tpokorra** [comment]
@Nicknag does this solution work?
Or do you need a parameter if it is not /wiki but /something else?

**Comment by Nicknag** [comment]
Hi @tpokorra

sorry for the late feedback. I have tried the solution but it does not work. I am not sure if this is due to a strange setup or rewrite rules on our side.

The thing is that I need to use https://mydomain.com/mediawiki/ as url but the links start with https://mydomain.com/wiki/. Not enough with this I also find links with "index.php?title=" for example the login link.

So what I did is to provide the complete base url for links as an optional parameter "linkurl" and then do the replace like this:

```
if linkurl:
        content = content.replace(f'a href="{linkurl}', f'a href="{url_title}')
```

`python exportMediaWiki2Html.py --url https://myurl.com/mediawiki/ --linkurl https://myurl.com/wiki/ --user myuser@htmlexport --password thisismypassword`
